python/sync-with-algolia/src/main.py

```diff
@@ -0,0 +1,77 @@
+import os
+from algoliasearch.search_client import SearchClient
+from appwrite.client import Client
+from appwrite.services.databases import Databases
+from appwrite.query import Query
+from utils import get_static_file, throw_if_missing, interpolate
+
+
+def main(context):
+    throw_if_missing(
+        os.environ,
+        [
+            "APPWRITE_API_KEY",
+            "APPWRITE_DATABASE_ID",
+            "APPWRITE_COLLECTION_ID",
+            "ALGOLIA_APP_ID",
+            "ALGOLIA_INDEX_ID",
+            "ALGOLIA_ADMIN_API_KEY",
+            "ALGOLIA_SEARCH_API_KEY",
+        ],
+    )
+
+    if context.req.method == "GET":
+        html = interpolate(
+            get_static_file("index.html"),
+            {
+                "ALGOLIA_APP_ID": os.environ["ALGOLIA_APP_ID"],
+                "ALGOLIA_INDEX_ID": os.environ["ALGOLIA_INDEX_ID"],
+                "ALGOLIA_SEARCH_API_KEY": os.environ["ALGOLIA_SEARCH_API_KEY"],
+            },
+        )
+        return context.res.send(html, 200, {"Content-Type": "text/html; charset=utf-8"})
+
+    client = (
+        Client()
+        .set_endpoint(os.environ["APPWRITE_ENDPOINT"] or "https://cloud.appwrite.io/v1")
+        .set_project(os.environ["APPWRITE_FUNCTION_PROJECT_ID"])
+        .set_key(os.environ["APPWRITE_API_KEY"])
+    )
+    database = Databases(client)
+
+    algolia = SearchClient.create(
+        os.environ["ALGOLIA_APP_ID"], os.environ["ALGOLIA_ADMIN_API_KEY"]
+    )
+    index = algolia.init_index(os.environ["ALGOLIA_INDEX_ID"])
+
+    cursor = None
+    while True:
```

Comment:
Spaces dont seem correct. Syncing logic (save_object) seems to be inside `while True`. I might be wrong tho, but please check it.